MapboxCoreNavigation/RouteLegProgress.swift

```diff
@@ -0,0 +1,255 @@
+
+import Foundation
+import MapboxDirections
+
+/**
+ `RouteLegProgress` stores the user’s progress along a route leg.
+ */
+open class RouteLegProgress: Codable {
```

Comment:
Thanks for splitting these classes out into their own files. Were there any changes to them other than adding the Codable-related methods?

Reply:
No other changes made.